src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/StronglyTypedEnums.txt

```diff
@@ -0,0 +1,52 @@
+﻿#SETUP: StronglyTypedBuiltinEnums,AllEnumsSetup
+
+>> ErrorKind.Div0 = Color.Red
+Errors: Error 15-16: Incompatible types for comparison. These types can't be compared: OptionSetValue(ErrorKind), OptionSetValue(Color).
+
+>> Color.Red
+Color.Red
+
+// Both string backed enums, but incompatble for comparison
+>> TimeUnit.Days <> JSONFormat.IndentFour
+Errors: Error 14-16: Incompatible types for comparison. These types can't be compared: OptionSetValue(TimeUnit), OptionSetValue(JSONFormat).
+
+// Valid to compare to strings
+>> TimeUnit.Days <> "abc"
+true
+
+// Coercion to Color
+>> ColorFade(Color.Red, 0)
+RGBA(255,0,0,1)
+
+>> Color.Red < Color.Green
+Errors: Error 10-11: Unable to compare values of type OptionSetValue(Color).
+
+// Number backed enums support gt/lt/geq/leq, but enforce type checks
+>> With({ErrorSev: ErrorSeverity.Warning}, ErrorSev < ErrorSeverity.Severe) 
+true
+
+>> ErrorSeverity.Warning < 5
+true
+
+>> ErrorSeverity.Warning = 5
+false
+
+>> With({ErrorSev: ErrorSeverity.Warning}, ErrorSev < StartOfWeek.Saturday) 
+Errors: Error 49-50: Incompatible types for comparison. These types can't be compared: OptionSetValue(ErrorSeverity), OptionSetValue(StartOfWeek).
+
+// Coercion to string uses backing values if the enum is string-backed
+>> JSONFormat.IndentFour & JSONFormat.IncludeBinaryData
+"4B"
+
+// Non string-backed enums use the name when coerced to string?
+// This might not be great, but it matches the CDS Option Set behavior?
+// We could block this or use the stringified backing value instead but it'd be a bit hackier
+>> "Color: " & Color.Red
+"Color: Red"
+
+// Functions can enforce expecting an enum type
```

Comment:
>can enforce expecting an enum type [](http://example.com/codeflow?start=13&length=34)

While I love this change, this may be a rather large breaking change that I don't think we can take

Reply:
Yes, this'll need to be rolled into the 1.0 flag, unfortunately we can't fix this up with a converter :(